### PR TITLE
chore: get `./gradlew format` to pass for the whole repo

### PR DESCRIPTION
**Pull request by postamar**
In principle this is the same thing as https://github.com/airbytehq/airbyte/pull/29686 but extended to the whole repo. The purpose of this PR is twofold:
1. educate myself on the different code formatters that we use or intend to use;
2. if possible identify a java code style which is both pleasant (see https://github.com/airbytehq/airbyte/pull/29686#discussion_r1300688583 for concerns about the existing style) and which isn't too different so as to not pollute the git history with more noise than necessary.

In the process of (1) I found that there's not much work required to get the `format` gradle task to succeed everywhere. This surfaced a few bugs which are perhaps worth fixing.

W.r.t (2) I found that https://github.com/palantir/palantir-java-format is tolerable if we tell it to conform to the Google style. This shouldn't be too surprising as it was derived from the latter originally. It's basically the same thing but with nicer handling of java-8+ lambdas. It seems to have more sophisticated rules than what's possible with the Eclipse CodeFormatter that we currently use. Styles are subjective and can be endlessly bike-shedded but I found the result to be quite nice. 

The first 3 commits actually have interesting changes, the rest are mechanical.

### CODE REVIEW

File: airbyte-integrations/connectors/source-confluence/metadata.yaml [hunk 1]
```diff
@@ -25,5 +25,4 @@ data:
   ab_internal:
     sl: 200
     ql: 300
-  supportLevel: community
```
Comment (postamar) [inline airbyte-integrations/connectors/source-confluence/metadata.yaml:28]: This key is duplicated and the other entry's value is `certified`. Which one is correct? I did this change assuming that `certified` overrides `community`.

Comment (postamar) [reply airbyte-integrations/connectors/source-confluence/metadata.yaml:28]: Note that in [the connector report](https://connectors.airbyte.com/files/generated_reports/connector_registry_report.html) the Confluence source shows up as `community`.

**Comment by cgardens** [comment]
@postamar for #2 is there are a particular issue you have with the existing formatter? I definitely agree, we need to run it for it be useful. Fwiw, the whole platform code base uses the existing formatter. It is not a hard requirement that we stay consistent across the two code bases, but I think we should default to doing that unless we have a strong reason to go in different directions.

**Comment by postamar** [comment]
Closing this in favour of https://github.com/airbytehq/airbyte/pull/29786 which is the same thing, but using the existing style. I'll submit a separate PR with the palantir style and we can continue the discussion there @cgardens.

**Pull request closed without merging**